x-pack/plugins/inference/common/chat_complete/errors.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { InferenceTaskError } from '../errors';
+import type { UnvalidatedToolCall } from './tools';
+
+export enum ChatCompletionErrorCode {
+  TokenLimitReachedError = 'tokenLimitReachedError',
+  ToolNotFoundError = 'toolNotFoundError',
+  ToolValidationError = 'toolValidationError',
+}
```

Comment:
Thoughts on rate limiting errors? Maybe provide an option to fail immediately, or after x number of retries? Or push this to inference API?

Reply:
Rate limiting errors is a good one, I will add those. I've left out retries for now, [it's fairly easy to implement for the consumer](https://rxjs.dev/api/operators/retry) and I suspect retry strategies might be a little more complicated rather than just repeating the same operation again - e.g. you might (probably?) want to send the error message back to the LLM to inform its next try.

Reply:
fwiw, I understood Garrett's more about handling rate limit errors from the external API, in the sense that we should throw a recognizable, normalized error, and possibly automatically retry.

Reply:
I will handle this in a follow-up - it's a little easier to manage if I have a real use case to generate these exceptions.

Reply:
++, this API however only handles a single interaction with the LLM, meaning the LLM can request tools to be called, but they're not actually called. That's out of scope for now. For the Observability AI Assistant, we have built-in tool call rate-limiting (it can only call n tools without intervention from the user).

Reply:
that sounds great, thank you for clarifying! once we get further along, it will be good to discuss ideas for how to generalize what o11y is doing here as well, so that as teams add new tools, rate limiting is taken into account from the beginning.